src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/factory/Cas1SpaceBookingEntityFactory.kt

```diff
@@ -129,6 +134,22 @@ class Cas1SpaceBookingEntityFactory : Factory<Cas1SpaceBookingEntity> {
     this.keyWorkerAssignedAt = { keyWorkerAssignedAt }
   }
 
+  fun withCancellationOccurredAt(occurredAt: LocalDate?) = apply {
+    this.cancellationOccurredAt = { occurredAt }
+  }
+
+  fun withCancellationRecordedAt(recordedAt: Instant?) = apply {
```

Comment:
Could you remind me the difference between Occured at and recorded at?
Why do these need to be two separate rows?
I wonder if there is a way to make it more obvious why the two times are required, as it may not be obvious to someone new to the codebase

Reply:
On the UI we ask the user to define the date that the cancellation occurred, which may not necessarily be the same as when it was entered into the system. It's never been clear to me why these would be different - I suspect it was to help in the early days when 'migrating' existing bookings into the system.

I'm pro comment if the code doesn't speak for itself so i've added a comment.

When we get to the wireframes for space booking cancellation we should double check if we actually still need to capture this and maybe remove it